samples/listenForAvroRecords.js

```diff
@@ -0,0 +1,107 @@
+// Copyright 2019-2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * This application demonstrates how to perform basic operations on
+ * subscriptions with the Google Cloud Pub/Sub API.
+ *
+ * For more information, see the README.md under /pubsub and the documentation
+ * at https://cloud.google.com/pubsub/docs.
+ */
+
+// This is a generated sample. Please see typescript/README.md for more info.
+
+'use strict';
+
+// sample-metadata:
+//   title: Listen For Avro Records
+//   description: Listens for records in Avro encoding from a subscription.
+//   usage: node listenForAvroRecords.js <subscription-name> [timeout-in-seconds]
+
+// [START pubsub_subscribe_avro_records]
+/**
+ * TODO(developer): Uncomment these variables before running the sample.
+ */
+// const subscriptionName = 'YOUR_SUBSCRIPTION_NAME';
+// const timeout = 60;
+
+// Imports the Google Cloud client library
+const {PubSub, Schema, Encodings} = require('@google-cloud/pubsub');
+
+// Node FS library, to load definitions
+const fs = require('fs');
+
+// And the Apache Avro library
+const avro = require('avro-js');
+
+// Creates a client; cache this for further use
+const pubSubClient = new PubSub();
+
+function listenForAvroRecords(subscriptionName, timeout) {
+  // References an existing subscription
+  const subscription = pubSubClient.subscription(subscriptionName);
+
+  // Make an encoder using the official avro-js library.
+  const definition = fs
+    .readFileSync('system-test/fixtures/provinces.avsc')
+    .toString();
+  const type = avro.parse(definition);
+
+  // Create an event handler to handle messages
+  let messageCount = 0;
+  const messageHandler = async message => {
+    // "Ack" (acknowledge receipt of) the message
+    message.ack();
+
+    // Get the schema metadata from the message.
+    const schemaMetadata = Schema.metadataFromMessage(message.attributes);
+
+    let result;
+    switch (schemaMetadata.encoding) {
+      case Encodings.Binary:
+        result = type.fromBuffer(message.data);
+        break;
+      case Encodings.Json:
+        result = type.fromString(message.data.toString());
+        break;
+    }
```

Comment:
Should also handle an all-else case?

Reply:
I went ahead and added default cases for all of these.